pb/datadeal/v0/deal.go

```diff
@@ -0,0 +1,23 @@
+package v0
+
+import (
+	"fmt"
+
+	"github.com/medibloc/panacea-oracle/panacea"
+)
+
+func (r *ValidateDataRequest) ValidateBasic() error {
+	if _, err := panacea.GetAccAddressFromBech32(r.ProviderAddress); err != nil {
+		return fmt.Errorf("invalid provider address: %w", err)
+	}
+
+	if len(r.EncryptedData) == 0 {
+		return fmt.Errorf("encrypted data is empty in request")
+	}
+
+	if len(r.DataHash) == 0 {
+		return fmt.Errorf("data hash is empty in request")
+	}
+
+	return nil
+}
```

Comment:
It's recommended for a `pb` directory to contain only auto-generated stubs, because we are usually gonna delete the entire `pb` directory before running `protoc`. If not, the directory may contain some old stubs that should be deleted according to the new proto definitions.
So, if possible, it would be good to move this file to somewhere else.

Reply:
Unfortunately, it seems that only the same package can declare functions for structs.
I think we can change this validation method so that it doesn't depend on the struct.

Reply:
Yes. I think we don't need to embed this function to the struct, since we don't need to follow the spec of Tendermint messages that should be callable by Cosmos SDK.

Reply:
Thank you. I fixed it.